src/components/AssetDescriptionComponent.jsx

```diff
@@ -8,23 +8,61 @@ import ConfirmAction from './common/ConfirmAction';
 import AssignedTo from './AssignAssetComponent';
 import '../_css/AssetDescriptionComponent.css';
 
-const AssetDescriptionComponent = (props) => {
+class AssetDescriptionComponent extends React.Component {
+ state = {
+   assignedUser: {},
+   assignAssetButtonState: true,
+   selectedUser: 0
+ }
+ onSelectUserEmail = (event, data) => {
+   this.setState({ selectedUser: data.value, assignAssetButtonState: false });
+ };
+
+handleAssign = () => {
+  const { selectedUser } = this.state;
+  const { id } = this.props.assetDetail;
+  const assetAllocated = {
+    asset: id,
+    current_owner: selectedUser
+  };
+  this.props.allocateAsset(assetAllocated, this.state.serialNumber);
+};
+
+handleUnassign = () => {
+  const { id } = this.props.assetDetail;
+  const assetAssigned = {
+    asset: id,
+    current_status: 'Available'
+  };
+  this.props.unassignAsset(assetAssigned, this.props.serialNumber);
+};
+
+handleConfirm = () => {
+  if (isEmpty(values(this.state.assignedUser))) {
+    return this.handleAssign();
+  }
+  return this.handleUnassign();
+};
+triggerProps = () => {
   let triggerProps = {
     buttonName: 'Assign Asset',
     customCss: 'assign-asset',
-    disabledState: props.assignAssetButtonState,
+    disabledState: this.state.assignAssetButtonState,
     color: 'primary'
   };
-
-  if (!isEmpty(values(props.assignedUser))) {
+  if (!isEmpty(values(this.state.assignedUser))) {
     triggerProps = {
-      ...triggerProps,
       buttonName: 'Unassign Asset',
       customCss: 'unassign-asset',
       disabledState: false
     };
-  }
-
+    return triggerProps;
+  } return triggerProps;
```

Comment:
Separate the brace from the return statement. I.e.
```
}
return triggerProps;

Reply:
Note that if you use the snippet in https://github.com/AndelaOSP/art-dashboard/pull/138#discussion_r219737669 then you don't have to action this comment